### PR TITLE
Fix json escaped character handling

### PR DESCRIPTION
**Pull request by Rambatino**
When cleanString was called on the data, it removed the `\\` which meant the escaping inside the JSON was then wrong. 

Unfortunately, the use-case supporting the reason for this cleaning wasn't codified in a test and thus I'm not 100% sure what its purpose was.

* Also fix the go mod version as was erroring when building locally
* Added a cancellation context to main.go and so SIGTERM now terminates the server

### CODE REVIEW

File: cmd/axiom-syslog-proxy/main.go [hunk 1]
```diff
@@ -38,7 +40,15 @@ func run(_ context.Context, _ *zap.Logger, client *axiom.Client) error {
...
+	defer stop()
 
-	return nil
+	go func() {
+		srv.Run()
+	}()
+
+	<-ctx.Done()
+
+	return ctx.Err()
```
Comment (lukasmalkmus) [inline cmd/axiom-syslog-proxy/main.go:53]: I don't think this change is needed. `github.com/axiomhq/pkg/cmd` deals with creating a proper cancellation context. Did you experience issues killing the binary?

Comment (Rambatino) [reply cmd/axiom-syslog-proxy/main.go:53]: Yeah couldn't kill it and couldn't kill the docker one either

Comment (lukasmalkmus) [reply cmd/axiom-syslog-proxy/main.go:53]: Either pass the given context to `srv` or just run it in a go routine and wait for it, like you're doing at the moment but just don't create a new `NotifyContext`. The first approach would be way cleaner, tho :)

There is a `Shutdown(ctx)` method on Go's http server + some additional tricks. AFAIK, the `github.com/axiomhq/pkg` might have a speciality http server package, if you wanna go that route ;) 

But that's extra credit, not mandatory :D Just don't create a second `NotifyContext`.

File: server/server.go [hunk 2]
```diff
@@ -110,19 +110,19 @@ func (srv *Server) Flush() error {
...
 
 	srv.started = true
 	ticker := time.NewTicker(5 * time.Second)
-	done := make(chan bool)
 
 	for {
 		select {
-		case <-done:
+		case <-ctx.Done():
+			srv.Flush()
```
Comment (lukasmalkmus) [inline server/server.go:125]: I think you wanna close the underlying network connections. I forgot this is not http, during my latest review :)

E.g. you wanna call `srv.tcpParser.Close()` and the same for the ump connection.

**Review by lukasmalkmus — APPROVED** [review]
LGTM

One nitpick.

**Review by lukasmalkmus — APPROVED** [review]
LGTM

**Pull request merged**